pkgs/development/tools/literate-programming/noweb/default.nix

```diff
@@ -1,29 +1,74 @@
-{stdenv, fetchurl, gawk}:
-
-stdenv.mkDerivation {
-  name = "noweb-2.11b";
-  src = fetchurl {
-    urls = [ "http://ftp.de.debian.org/debian/pool/main/n/noweb/noweb_2.11b.orig.tar.gz"
-             "ftp://www.eecs.harvard.edu/pub/nr/noweb.tgz"
-          ];
-    sha256 = "10hdd6mrk26kyh4bnng4ah5h1pnanhsrhqa7qwqy6dyv3rng44y9";
+{ stdenv, fetchFromGitHub, gawk, gcc, groff, icon-lang ? null }:
+
+let noweb = stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
+  version = "2.12";
+  pname = "noweb";
+  tlType = "run";
+
+  src = fetchFromGitHub {
+    owner = "nrnrnr";
+    repo = "noweb";
+    rev = "v${builtins.replaceStrings ["."] ["_"] version}";
+    sha256 = "1160i2ghgzqvnb44kgwd6s3p4jnk9668rmc15jlcwl7pdf3xqm95";
   };
+
+  outputs = [ "out" "tex" ];
+
+  nativeBuildInputs = [ gcc groff ] ++ stdenv.lib.optionals (!isNull icon-lang) [ icon-lang ];
```

Comment:
Is `gcc` actually required here?

Reply:
I think we get it from `stdenv`, so perhaps not. I will test.